device/cuda/src/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.cu

```diff
@@ -364,9 +364,10 @@ greedy_ambiguity_resolution_algorithm::operator()(
     m_copy.get().setup(block_offsets_buffer)->ignore();
     vecmem::data::vector_buffer<int> scanned_block_offsets_buffer{nBlocks_scan,
                                                                   m_mr.main};
-    m_copy.get().setup(block_offsets_buffer)->ignore();
+    m_copy.get().setup(scanned_block_offsets_buffer)->ignore();
 
-    while (!terminate && n_accepted > 0) {
+    unsigned int iter = 0;
+    while (!terminate && n_accepted > 0 && iter < m_config.max_iterations) {
```

Comment:
Hmm... If we want to make a big loop "out here", then the setup of the CUDA graph needs to be taken outside of this loop. Since as long as we keep re-making the graph on each iteration, the graph code will just slow us down.

So I'd re-think this part of the code even more fundamentally. 🤔

Reply:
@krasznaa is correct. The graph is not supposed to be created every iteration. Probably you are not seeing the impact due to small size of data.

Even though it is ideal to never re-created the cudaGraph, currently the cudaGraph is recreated every 100 iterations to update the block and grid size based on the number of remaining tracks.